### DISCUSSION TITLE
SuperSlicer adding incorrect/unwanted gcode

### DISCUSSION THREAD

**Discussion by phongshader** [supermerill/SuperSlicer#2816] [Q&A]
I'm new to SuperSlicer so go easy if I'm asking a stupid question. SS is adding a line in my gcode, `SET_PRESSURE_ADVANCE ADVANCE=0` which throws an error in Klipper, `Unable to infer active extruder stepper` because no extruder is specified. This is a 2 in 1 out printer. The correct code should be `SET_PRESSURE_ADVANCE ADVANCE=0 EXTRUDER=<tool name>`. The 2 places I've added gcode in SS is Printer settings > Custon G-code > Start G-code > `SET_PRESSURE_ADVANCE ADVANCE=0 EXTRUDER=[tool_name]`
and Filament Settings > Custom G-code > Start G-code > `SET_PRESSURE_ADVANCE ADVANCE=<x.x> SMOOTH_TIME=<x.x> EXTRUDER=[tool_name]`. Other than that I don't know where else to look to see where `SET_PRESSURE_ADVANCE ADVANCE=0` is being added to the gcode. I tried removing `SET_PRESSURE_ADVANCE ADVANCE=0 EXTRUDER=[tool_name]` from the printer start gcode but that made no difference. I tried removing all custom gcode and the command was still there. Any Ideas, suggestions?

**Comment by phongshader** [comment]
Apparently this is a problem introduced with the deprecation of `SHARED_HEATER` in klipper.  It is discussed here https://github.com/supermerill/SuperSlicer/issues/2599

**Reply by supermerill** [reply]
That illustrate the need to have a configurable gcode writer so users can script what to write for these kind of very specific things.
Sadly, it's a big task.

In the 2.4, you may be able to use the new gcode "search & replace" thing prusa added to patch it.

**Comment by supermerill** [comment]
what is the vendor profile that does that? voron?

**Reply by phongshader** [reply]
custom profile, Beast V2 printer
